File: programmers/javascript/level2/기능개발.js
const temp1 = [93, 30, 55];
const temp2 = [1, 30, 5];

const solution = (progresses, speeds) => {
  let temp = [];
  const answer = [];
  temp = progresses.map((it, i) => Math.ceil((100 - it) / speeds[i]));
  let startNum = temp[0];
  let num = 1;
  for (let i = 0; i < temp.length; i += 1) {
    if (startNum >= temp[i + 1] && i !== temp.length - 1) {
      num += 1;
    } else {
      answer.push(num);
      startNum = temp[i + 1];
      num = 1;
    }
  }
  return answer;
};

console.log(solution(temp1, temp2));

File: python/programmers/level2/기능개발.py
import math


def solution(progresses, speeds):
    answer = []
    tmp = []
    for p, s in zip(progresses, speeds):
        tmp.append(math.ceil((100 - p) / s))
    count = 1
    num = tmp[0]
    for i in range(1, len(tmp)):
        if num < tmp[i]:
            answer.append(count)
            count = 1
            num = tmp[i]
        else:
            count += 1
    answer.append(count)
    return answer


print(solution([95, 90, 99, 99, 80, 99], [1, 1, 1, 1, 1, 1]))


# def solution(progresses, speeds):
#     Q = []
#     for p, s in zip(progresses, speeds):
#         if len(Q) == 0 or Q[-1][0] < -((p - 100) // s):
#             Q.append([-((p - 100) // s), 1])
#         else:
#             Q[-1][1] += 1
#     return [q[1] for q in Q]
